utils/s3_csv_reader.py

```diff
@@ -5,7 +5,7 @@
 
 CACHE_MANAGER = CacheManager(verbose=False)
 
-def read_csv(filename: str | Path) -> pd.DataFrame:    
+def read_csv(filename: str | Path) -> pd.DataFrame:
```

Comment:
I get an error here -- is this a Python version issue? I'm on 3.9.7, I'm using a fresh virtual environment with everything from `requirements.txt` installed so hopefully not a package issue?

```
  File "/Users/laurie/git/hailey-fork-ghost-buses/utils/s3_csv_reader.py", line 8, in <module>
    def read_csv(filename: str | Path) -> pd.DataFrame:
TypeError: unsupported operand type(s) for |: 'type' and 'type'
```

Reply:
(Temporarily resolved by just deleting the `| Path` so I can run)

Reply:
Yes, I believe it's a Python version issue. I've been testing everything with 3.11.8. We can look more closely at the minimum Python version we want to support.

Reply:
read_csv doesn't need to take a string, so I simplified it slightly, which should also make this run in earlier Python versions.